Newsletter vol.4

syleeeee opened the pull request:
- Add the newsletter vol. 4

Discussion items
- Should we create some more good first issues? [Currently, all of them are WIP](https://github.com/line/armeria/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22).
- I just added all of the release note summaries, but should we merge and display them in small groups? Such as `New support`, `gRPC new features`, `Fixes`, etc ..

Any other comments are appreciated :) 

![screencapture-localhost-8000-news-20220708-newsletter-4-2022-07-07-17_13_40](https://user-images.githubusercontent.com/6274210/177726479-1812367b-b80a-483b-8ac0-042c77347dc7.png)

minwoox commented on site/src/pages/news/20220708-newsletter-4.mdx:
```diff
@@ -0,0 +1,156 @@
+  - Exception handler
+  - Blocking task executor
+- 🦾 MSA(Microservices Architecture): Harder, Better, Faster, Stronger
+  - In the session, @cnabro introduces how the "Series On" service converted monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentioned about how they monitored microservices using Zipkin and Elastic Search.
```
```suggestion
  - In the session, @cnabro introduces how [Series On](https://serieson.naver.com/) migrated their monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentioned how they monitored microservices using [Zipkin](https://zipkin.io/) and [Elasticsearch](https://www.elastic.co/).
```

jrhee17 replied:
The tense should be consistent (introduces <-> mentions)

```suggestion
  - In the session, @cnabro introduces how the "Series On" service converted monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentions how they monitored microservices using Zipkin and Elastic Search.
```

minwoox commented:
> Should we create some more good first issues?

Let us create more issues before we send this letter. 😉 

> I just added all of the release note summaries, but should we merge and display them in small groups?

I prefer the current way except for putting the latest release at the top. 😄

jrhee17 commented on site/src/pages/news/20220708-newsletter-4.mdx:
```diff
@@ -0,0 +1,156 @@
+<div class="noBullet">
+
+- 👂 Armeria team is hosting [bi-weekly virtual office hours](https://armeria.dev/community/office-hours/).
+  - It was great talking with all of you through the video chat. And we're still listening. :)
```
I think the sentence sounds a little vague. Did you mean we're still doing virtual hours?

```suggestion
  - It was great talking with all of you through video chat. We hope we can see you again in the near future. :)
```

ikhoon commented on site/src/pages/news/20220708-newsletter-4.mdx:
```diff
@@ -0,0 +1,156 @@
+
+<div class="noBullet">
+
+- ✨ We released several releases since our last newsletter with a bunch of new exciting features!
```
According to [Amazon's writing style tips](https://medium.com/fact-of-the-day-1/amazon-writing-style-tip-2-2f77bb3bbbb0), it recommends replacing adjectives with numbers. 😆
```suggestion
- ✨ We've released 7 new versions since our last newsletter with a bunch of new exciting features!
```

jrhee17 commented on site/src/pages/news/20220708-newsletter-4.mdx:
```diff
@@ -0,0 +1,156 @@
+
+<div class="noBullet">
+
+- 🎓 Try out this guided tutorial for REST Service](https://armeria.dev/tutorials/rest/blog)! It's super simple, and super friendly to understand building a RESTful service with Aremria. This tutorial will guide you to make a blog service, with which you can create, read, update and delete blog posts. In particular, you'll be using these Armeria features:
```
- Shouldn't the link be on `tutorial` instead of `REST Service`?
- Prefer active voice, "It" is vague in this context
- Removed a redundant sentence

```suggestion
- 🎓 Try out this guided [tutorial](https://armeria.dev/tutorials/rest/blog) for creating a RESTful service! This super friendly and simple tutorial will guide you to make a blog service, with which you can create, read, update and delete blog posts. In particular, you'll be using these Armeria features:
```

jrhee17 commented on site/src/pages/news/20220708-newsletter-4.mdx:
```diff
@@ -0,0 +1,156 @@
+  - Blocking task executor
+- 🦾 MSA(Microservices Architecture): Harder, Better, Faster, Stronger
+  - In the session, @cnabro introduces how the "Series On" service converted monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentioned about how they monitored microservices using Zipkin and Elastic Search.
+  - Here are the [Video](https://youtu.be/1TsGYR1dves) and [Slide](https://speakerdeck.com/line_devday2021/msa-microservices-architecture-harder-better-faster-stronger) (also providing [Korean](https://youtu.be/95L-_82N1vg) and [Japanese](https://youtu.be/pXIdcgchvSU))
```
Since the subject "Here" is singular

```suggestion
  - Here is the [Video](https://youtu.be/1TsGYR1dves) and [Slide](https://speakerdeck.com/line_devday2021/msa-microservices-architecture-harder-better-faster-stronger) (also providing [Korean](https://youtu.be/95L-_82N1vg) and [Japanese](https://youtu.be/pXIdcgchvSU))
```

jrhee17 commented on site/src/pages/news/20220708-newsletter-4.mdx:
```diff
@@ -0,0 +1,156 @@
+  - [1.15.0](https://armeria.dev/release-notes/1.15.0/)
+    - User-friendly gRPC health check protocol
+    - Multipart file uploads in annotated services
+    - Preliminary GraphQL support in <type://DocService>
```
Since preliminary sounds like it's not ready for usage, but it is 😄 

```suggestion
    - GraphQL support in <type://DocService>
```

jrhee17 commented on site/src/pages/news/20220708-newsletter-4.mdx:
```diff
@@ -0,0 +1,156 @@
+    - User-friendly gRPC health check protocol
+    - Multipart file uploads in annotated services
+    - Preliminary GraphQL support in <type://DocService>
+    - Refactored DMS resolver cache with higher hit ratio and reduced traffic
```
```suggestion
    - Refactored DNS resolver cache with higher hit ratio and reduced traffic
```

jrhee17 commented on site/src/pages/news/20220708-newsletter-4.mdx:
```diff
@@ -0,0 +1,156 @@
+  - [1.14.1](https://armeria.dev/release-notes/1.14.1/)
+    - Additional logs for <type://EmptyEndpointGroupException>
+  - [1.15.0](https://armeria.dev/release-notes/1.15.0/)
+    - User-friendly gRPC health check protocol
```
I'm not sure if this additional support is particularly use friendly

```suggestion
    - Support for gRPC health check protocol
```

jrhee17 reviewed (commented):
Overall looks good 👍  Left some minor comments

syleeeee commented:
Thanks for reviewing. I addressed your comments.
![screencapture-localhost-8000-news-20220714-newsletter-4-2022-07-14-16_24_43](https://user-images.githubusercontent.com/6274210/178926088-375d0dad-e8f7-4d69-8f26-144e194f0540.png)

ikhoon commented on site/src/pages/news/20220714-newsletter-4.mdx:
````diff
@@ -0,0 +1,169 @@
+🔰 Want to contribute but not sure where to start from? Try one of these:
+  - #4237 Support for request scoped DataLoader for GraphQL-Java
+  ```java
+  class GraphqlServiceBuilder {
+      public GraphqlServiceBuilder dataLoaderRegistry(
+        Function<ServiceRequestContext,
+        DataLoaderRegistry>dataLoaderRegistrySupplier) {
+           // This method will be mutually exclusive
+           // with configureDataLoaderRegistry(...)
+      }
+  }
````
```suggestion
  class GraphqlServiceBuilder {
    public GraphqlServiceBuilder dataLoaderRegistry(
      Function<ServiceRequestContext,
        DataLoaderRegistry> dataLoaderRegistrySupplier) {
          // This method will be mutually exclusive
          // with configureDataLoaderRegistry(...)
    }
  }
```

jrhee17 commented on site/src/pages/news/20220714-newsletter-4.mdx:
````diff
@@ -0,0 +1,169 @@
+      }
+  }
+  ```
+  - #4347 Show the full stack trace of <type://CompositeException> when the system property verboseExceptions is `true`
````
```suggestion
  - #4347 Show the full stack trace of <type://CompositeException> when the system property `verboseExceptions` is `true`
```

jrhee17 commented on site/src/pages/news/20220714-newsletter-4.mdx:
```diff
@@ -0,0 +1,169 @@
+  - Blocking task executor
+- 🦾 MSA(Microservices Architecture): Harder, Better, Faster, Stronger
+  - In the session, @cnabro introduces how [Series On](https://serieson.naver.com/) migrated their monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentions about how they monitored microservices using [Zipkin](https://zipkin.io/) and [Elasticsearch](https://www.elastic.co/).
+  - Here is the [Video](https://youtu.be/1TsGYR1dves) and [Slide](https://speakerdeck.com/line_devday2021/msa-microservices-architecture-harder-better-faster-stronger) (also providing [Korean](https://youtu.be/95L-_82N1vg) and [Japanese](https://youtu.be/pXIdcgchvSU))
```
Oops, I didn't see the entire sentence correctly 😅 Let me revert this change for you.

```suggestion
  - Here are the [Video](https://youtu.be/1TsGYR1dves) and [Slides](https://speakerdeck.com/line_devday2021/msa-microservices-architecture-harder-better-faster-stronger) (also providing [Korean](https://youtu.be/95L-_82N1vg) and [Japanese](https://youtu.be/pXIdcgchvSU))
```

syleeeee replied:
I trusted you more than I trusted myself 🤣

trustin commented on site/src/pages/news/20220714-newsletter-4.mdx:
```diff
@@ -0,0 +1,169 @@
+<div class="noBullet">
+
+- ✨ We've released 7 new versions since our last newsletter with a bunch of new exciting features!
+  - [1.17.1](https://armeria.dev/release-notes/1.17.1/)
+    - Fixing a bug where Retrofit client calls hang indefinitely
```
How about removing micro versions in the list, so we focus on new features only? I'm afraid people might stop reading if the list gets too large.

trustin commented on site/src/pages/news/20220714-newsletter-4.mdx:
```diff
@@ -0,0 +1,163 @@
+    - Easier way to send and receive RESTful APIs
+    - Ability to limit the maximum length of Thrift messages
+    - Injecting dependencies in annotations
+    - Support for Kotlin nullable type(`?`) in annotated services
```
```suggestion
    - Support for Kotlin nullable type (`?`) in annotated services
```

trustin reviewed (approved):
Just one nit - Thanks a lot, @syleeeee! 🙇

syleeeee commented:
Thanks all for reviewing :-) 
Newsletter vol.4 is published.

ikhoon commented:
I've received it! 😆 Thanks a lot!
